packages/edit-post/src/hooks/components/media-upload/index.js

```diff
@@ -199,7 +199,21 @@ class MediaUpload extends Component {
 		}
 	}
 
+	updateCollection() {
+		const frameContent = this.frame.content.get();
+		if ( frameContent ) {
+			const collection = frameContent.collection;
+			// clean all attachments we have in memory.
+			collection.toArray().forEach( ( model ) => model.trigger( 'destroy', model ) );
```

Comment:
Hi @antpb, If I'm not in error I think @aduth  suggested me during the tries of https://github.com/WordPress/gutenberg/pull/6390 that a possible improvement would be destroying the models when the modal Closes. This would have the advantage of only keeping models in memory when they are needed.

Reply:
@jorgefilipecosta thanks for that. Is the updated code more what you are saying here? I added a `destroyCollection` function to trigger in `onClose`. Doing the destroy on close and doing the `collection.more();` in updateCollection. 

In my local it seems to be working as intended. :)

Reply:
I was doing some testing of having the collection destroyed at `onClose` now when you select a media item using the Media Library button, it does not return back the data needed because onClose it is clearing the collection. I think we need to keep the process of destroying and requesting new items in `updateCollection`